src/formatters/utils/sortResults.ts

```diff
@@ -1,13 +1,79 @@
 import { IRuleResult } from '../../types';
 
-export const sortResults = (results: IRuleResult[]) => {
-  return [...results].sort((resultA, resultB) => {
-    const diff = resultA.range.start.line - resultB.range.start.line;
+const compareCode = (one: string | number | undefined, another: string | number | undefined): number => {
+  if (one === void 0 && another === void 0) {
+    return 0;
+  }
+
+  if (one === void 0) {
+    return -1;
+  }
+
+  if (another === void 0) {
+    return 1;
+  }
+
+  return one.toString().localeCompare(another.toString(), undefined, { numeric: true });
+};
+
+const compareSource = (one: string | undefined, another: string | undefined): number => {
+  if (one === void 0 && another === void 0) {
+    return 0;
+  }
+
+  if (one === void 0) {
+    return -1;
+  }
 
-    if (diff === 0) {
-      return resultA.range.start.character - resultB.range.start.character;
-    }
+  if (another === void 0) {
+    return 1;
+  }
+
+  return one.localeCompare(another);
+};
+
+const normalize = (value: number): -1 | 0 | 1 => {
```

Comment:
```suggestion
const normalize = (value: number): -1 | 0 | 1 => Math.max(-1, Math.min(1, Math.floor(value))) as -1 | 0 | 1;
```

Math.floor is optional, but it's meant to protect us from returning other values than integers.

Reply:
@P0lip Hmmm. TBH, I do believe that this proposal is less easier to parse than the original (boring) code. I tend to avoid those one-liner smart constructs for readability purposes.

However, if you really want me to update the code, I'll do it, of course.

Reply:
Nah, I don't really mind. :)